### ISSUE TITLE
Fix the "location" keyword classification to refer to schemas only

### ISSUE THREAD

**Issue by handrews** [json-schema-org/json-schema-spec#1306]
The current "location" behavior was kind of a dumping ground for keywords that didn't have an immediate action associated with them, namely `$defs` and `$comment`.

However, _schema_ location behavior is critically important for proper detection of embedded resources and fragment declarations.  As a behavioral marker for extension keywords, it informs an implementation where to look for these things.  Regardless of what else we may or may not do with the current keyword classification section, we should fix this to be a more useful category.  Agreeing to do so does not preclude a more substantial change in this area later, once other discussions have resolved.

`$comment` can be described as having unique behavior.  If we eventually adopt a more elaborate keyword behaviors approach, it will fit into such a thing, but I don't want to worry about that much more complex topic right now.  It's not clear that we'll act on that idea at all so it shouldn't block this simpler fix.

**Comment by notEthan** [comment]
> `contentSchema` ... shouldn't be processed at schema load time like a subschema of an applicator or location ($defs) would be. In other words, you shouldn't be able to target it with a $ref even though it looks like a schema. It's just data. (@handrews)

I don't think this is the case in the current published spec - this issue is framed as a clarification, but it seems to me like a major change in how `contentSchema` is specified. Right now the spec says it's a schema, the metaschema says it's a schema, nothing indicates treatment any different than any other subschema like `$defs` values. (I'm sure there's volumes of issue comments that I could read that would lead me to understand your baseline assertion that it already is just data and not a schema, but that is sure not what's in the spec.)

I think it is a good change to make. Treating `contentSchema` as a subschema, it doesn't do much - no validation, no inplace application, it doesn't describe its instance (until that instance is parsed to something else). The only thing it does is what @handrews wants to prevent, `$ref`ing into it by anchor or by pointer, if it is a subschema without an $id.

I agree it should just be schema-shaped data, not a subschema. This does have other implications, though.

The metaschema currently describes `contentSchema` as a schema. I rely on the metaschema describing its subschemas for nearly everything that needs to know what is a schema and what isn't. If `contentSchema` isn't a subschema, the metaschema should not say it is. This, in my implementation, directly means it won't be recognized as a schema for anchor collection and $refing in will not find something that is a schema.

That affects validation: if the metaschema doesn't describe it as a schema, its content won't be validated as a schema if you validate against the metaschema. I think that is fine. It can be validated as a schema if/when its data are interpreted as a schema.

As just data, it should become a schema only considered as its own detached document, with the rules that apply to root schemas. This is a bit different than resource subschemas (subschemas with an $id).

- It is a resource schema, even without an $id.
  - However, lacking an $id has the problem @handrews noted that its retrieval URI is the same as its parent's, if present.
  - This implies its metaschema may be specified by $schema (regardless of $id).
- @jdesrosiers' example with `"contentSchema": { "$ref": "#/$defs/value" }` relative to the root of the resource containing the `contentSchema` won't work. The `contentSchema` *schema* has its root document equal to what's below `contentSchema`. It can reference the example's `value` schema by absolute URI or by moving it into `contentSchema`.

**Comment by channi23** [comment]
Hey @handrews 

I know this issue has been open for a while, so before drafting anything I wanted to check whether the intended direction here is still current.

My understanding is that the goal would be to clarify the keyword classification so that `Reserved Locations` refers specifically to locations reserved for schemas, such as `$defs`, while `$comment` would be described separately as having its own distinct behavior.

If that is still the right scope, I’d be interested in putting together a small PR against `specs/jsonschema-core.md` that keeps this limited to clarification/editorial cleanup rather than changing implementation behavior.
